addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/handler/KodiHandler.java

```diff
@@ -274,10 +275,19 @@ public void playURI(Command command) {
         connection.playURI(command.toString());
     }
 
-    public void playFavorite(Command command) {
-        String path = connection.getFavoritePath(command.toString());
-        if (StringUtils.isNotEmpty(path)) {
-            connection.playURI(path);
+    private void playFavorite(Command command) {
+        KodiFavorite favorite = connection.getFavorite(command.toString());
+        if (favorite != null) {
+            String path = favorite.getPath();
+            String windowParameter = favorite.getWindowParameter();
+            if (StringUtils.isNotEmpty(path)) {
+                connection.playURI(path);
+            } else if (StringUtils.isNotEmpty(windowParameter)) {
+                final String[] windowParameters = { windowParameter };
```

Comment:
A final string array is still mutable, I prefer not to use it in such cases because it could be misleading for less experienced developers

Reply:
What do you suggest in this case? Shall I use a `List` instead of an array?

Reply:
Oh, ok. I misunderstood. That should be no problem.